bin/server.js

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env node
+var debug             = require('debug')('github:server');
+var base              = require('taskcluster-base');
+var api               = require('../lib/api');
+var path              = require('path');
+var Promise           = require('promise');
+var exchanges         = require('../lib/exchanges');
+var _                 = require('lodash');
+
+/** Launch server */
+var launch = async function(profile, publisher) {
```

Comment:
So for pulse we could fake that with a vagrant setup... add pulse.mozilla.org to `/etc/hosts` and install rabbitmq locally. We don't need the security model to be enforced and could just use the rabbitmq default user for testing.

I'm not oppose to these things. But stubbing out the publisher means that you skip validation etc... and it very quickly becomes rather boring... So if we really want to do something like this, perhaps we should add an option to `base.Exchanges` such that a boolean config option can be used to get a dummy publisher that just validates messages and emits them as events on the `process` object.

We should probably do the same for `stats.Influx`, such that it looks like:
```js
statsDrain = new base.stats.Influx({
  connectionString:   cfg.get('influx:connectionString'),
  // then in config/test.js, set influx:connectionString = "null"
  // and modify base.stats.Influx such that connectionString means it will print stats as debug messages
  // and emit them as events on the process object.
  maxDelay:           cfg.get('influx:maxDelay'),
  maxPendingPoints:   cfg.get('influx:maxPendingPoints')
});
```

I guess we could have two test profiles in `config/` one for `local-testing` and `production-testing`. Either way, if we want to do this, we should make it nice :)

-------

So far the reasoning for not doing this have been that we didn't care. And that we prioritized getting things out the door. The golden standard will always be blackbox testing against production. Which is generally possible because we can make special test credentials that can only mess with `dummy-provisioner`, and things like that.